### PR TITLE
Redirect

### PR DESCRIPTION
**Pull request by tok-kkk**
**Description**

The PR introduce some changes to the darknode-cli which includes:

- `darknode up` redirects to Darknode Command Centre after deployed successfully.
- `darknode destroy` will check for deregistration (it is insufficient to be pending deregistration) before continuing
- `darknode destroy` redirects to Darknode Command Centre for Darknodes that are not deregistered
- Remove `force` tag for `darknode destroy` command. Cli will not continue running until the Darknode has been completely deregistered

**Unresolved Issues**

- There are some potential dependency issues. We are waiting for the `republic-go` to use higher  version of `go-ethereum`. Go 1.11 has issue with `go-ethereum 1.8.12` when compiling the cli.

### CODE REVIEW

File: cmd/down.go [hunk 1]
```diff
@@ -200,3 +240,27 @@ func renAddress(network contract.Network) string {
 		return ""
 	}
 }
+
+// darknode registry on different testnet
+func dnrAddress(network contract.Network) string {
```
Comment (divyakoshy) [inline cmd/down.go]: `beth-go`'s address-book can be used here to get dnrAddress for any given network.

File: cmd/down.go [hunk 2]
```diff
@@ -23,38 +26,79 @@ import (
...
+		time.Sleep(3 * time.Second)
+
+		var redirect *exec.Cmd
+		switch runtime.GOOS {
+		case "darwin":
+			redirect = exec.Command("open", fmt.Sprintf("https://darknode.republicprotocol.com/status/%v", ip))
+		case "linux":
+			redirect = exec.Command("xdg-open", fmt.Sprintf("https://darknode.republicprotocol.com/status/%v", ip))
+		default:
+			log.Fatal("unsupported operating system")
```
Comment (divyakoshy) [inline cmd/down.go]: Consider making this `ErrUnsupportedOS` or something similar and returning it.

File: cmd/down.go [hunk 3]
```diff
@@ -23,35 +26,72 @@ import (
...
-				return nil
-			}
+		var redirect *exec.Cmd
+		switch runtime.GOOS {
+		case "darwin":
+			redirect = exec.Command("open", fmt.Sprintf("https://darknode.republicprotocol.com/status/%v", ip))
+		case "linux":
+			redirect = exec.Command("xdg-open", fmt.Sprintf("https://darknode.republicprotocol.com/status/%v", ip))
+		default:
+			return errors.New("unsupported operating system")
```
Comment (divyakoshy) [inline cmd/down.go]: Consider naming this `ErrUnsupportedOS` or something similar.

File: cmd/up.go [hunk 4]
```diff
@@ -129,5 +131,20 @@ func outputUrl(name, nodeDir string) error {
...
+
+	// Redirect the user to the registering URL.
+	var redirect *exec.Cmd
+	switch runtime.GOOS {
+	case "darwin":
+		redirect = exec.Command("open", fmt.Sprintf("https://darknode.republicprotocol.com/status/%v", ip))
+	case "linux":
+		redirect = exec.Command("xdg-open", fmt.Sprintf("https://darknode.republicprotocol.com/status/%v", ip))
+	default:
+		log.Fatal("unsupported operating system")
```
Comment (divyakoshy) [inline cmd/up.go]: Consider using the `ErrUnsupportedOS` here as well.

File: cmd/down.go (hunk 1 above)
Comment (tok-kkk) [reply cmd/down.go]: The address book for mainnet is empty in `beth-go`

File: cmd/down.go (hunk 3 above)
Comment (tok-kkk) [reply cmd/down.go]: Good suggestion. Thanks.

File: cmd/up.go (hunk 4 above)
Comment (tok-kkk) [reply cmd/up.go]: Good suggestion. Thanks.

**Pull request merged**